pkg/kubelet/pod_workers.go

```diff
@@ -87,36 +95,39 @@ func newPodWorkers(runtimeCache kubecontainer.RuntimeCache, syncPodFn syncPodFnT
 		runtimeCache:              runtimeCache,
 		syncPodFn:                 syncPodFn,
 		recorder:                  recorder,
+		workQueue:                 workQueue,
+		resyncInterval:            resyncInterval,
+		backOffInterval:           backOffInterval,
 	}
 }
 
 func (p *podWorkers) managePodLoop(podUpdates <-chan workUpdate) {
 	var minRuntimeCacheTime time.Time
 	for newWork := range podUpdates {
-		func() {
-			defer p.checkForUpdates(newWork.pod.UID, newWork.updateCompleteFn)
+		func() (err error) {
+			defer func() { p.wrapUp(newWork.pod.UID, err) }()
```

Comment:
I had to think really hard (and reread Daniel's [go landmines](https://gist.github.com/lavalamp/4bd23295a9f32706a48f)) about whether this was scoped correctly. I believe it is, but I think it would be more readable to get the error returned by the function and call wrapup then. E.g.:

```go
err := func() (err error) {
   ...
}()
p.wrapUp(newWork.pod.UID, err)